CodeEdit/Features/CEWorkspace/Models/Resource.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  CEWorkspaceActor.swift
+//  CodeEdit
+//
+//  Created by Wouter Hennen on 19/06/2023.
+//
+
+import Foundation
+import AppKit
+import SwiftUI
+
+protocol Resource: AnyObject, Identifiable {
```

Comment:
I'm wondering on whether the term `Resource` is too arbitrary

Reply:
Feel free to suggest better names. I chose this because it's used a lot throughout the codebase and is memorizable (compared to, for example, CEWorkspaceFile)

Reply:
Apple's docs also refer to filesystem items as resources: https://developer.apple.com/documentation/foundation/urlresourcekey/1414512-fileresourceidentifierkey